core/src/main/java/discord4j/core/object/entity/MemberPart.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discord4j.core.object.entity;
+
+import discord4j.common.util.Snowflake;
+import discord4j.discordjson.json.PartialMemberData;
+import discord4j.discordjson.possible.Possible;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** A part of Discord guild member. */
+public class MemberPart {
```

Comment:
Let's collapse `MemberPart` into `UserWithMember` (or `PartialMember` after its renamed). All of these methods can be present directly there instead of having to go through `getMember()`.